app/src/main/java/com/example/myweatherapp/adapter/WeatherAdapter.kt

```diff
@@ -7,38 +7,71 @@ import androidx.recyclerview.widget.RecyclerView
+import com.example.myweatherapp.databinding.WeatherFragmentBinding
+import com.example.myweatherapp.utils.WeatherStringUtils
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+interface ItemClicked {
+    fun weatherClicked(date: String)
+}
+
+class WeatherAdapter(
+    private val weather: List<DaysWeather>,
+    private val item: ItemClicked? = null
+) :
+    RecyclerView.Adapter<WeatherAdapter.ViewHolder>() {
 
-class WeatherAdapter (private val weather: List<DaysWeather>) : RecyclerView.Adapter<WeatherAdapter.ViewHolder>() {
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
-        ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.weather_frag, parent, false))
+        ViewHolder(
+            LayoutInflater.from(parent.context).inflate(R.layout.weather_fragment, parent, false),
+            item
+        )
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        holder.bind(weather[position])
+        val item = weather[position]
+        holder.bind(item)
     }
 
     override fun getItemCount() = weather.size
 
-    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-        private val binding = WeatherFragBinding.bind(itemView)
+    class ViewHolder(itemView: View, private val item: ItemClicked?) :
+        RecyclerView.ViewHolder(itemView) {
+        private val binding = WeatherFragmentBinding.bind(itemView)
 
         fun bind(weatherItem: DaysWeather) {
             binding.apply {
-                actualTemp.text = "${weatherItem.temperature.temp}${itemView.context?.getString(R.string.celsius)}"
-                tempMax.text = "${weatherItem.temperature.tempMax}${itemView.context?.getString(R.string.celsius)}"
-                tempMin.text = "${weatherItem.temperature.tempMin}${itemView.context?.getString(R.string.celsius)}"
+                actualTemp.text =
+                    "${weatherItem.temperature.temp}${itemView.context?.getString(R.string.celsius)}"
+                tempMax.text =
+                    "${weatherItem.temperature.tempMax}${itemView.context?.getString(R.string.celsius)}"
+                tempMin.text =
+                    "${weatherItem.temperature.tempMin}${itemView.context?.getString(R.string.celsius)}"
                 realFeel.text =
-                    "${itemView.context?.getString(R.string.real_feel)}${weatherItem.temperature.feelsLike}${itemView.context?.getString(R.string.celsius)}"
-                date.text = weatherItem.date
+                    "${itemView.context?.getString(R.string.real_feel)}${weatherItem.temperature.feelsLike}${
+                        itemView.context?.getString(
+                            R.string.celsius
+                        )
+                    }"
+                if (item == null) {
+                    val formatMilitaryTime = SimpleDateFormat(WeatherStringUtils.JSON_MILITARY_TIME_PATTERN, Locale.ENGLISH)
```

Comment:
you have this repeated in line 61. please put this outside the if so you will not repeat code

Reply:
Done. thx

Reply:
Done thx